module/PowerShellEditorServices/Commands/Public/CmdletInterface.ps1

```diff
@@ -125,7 +125,15 @@ function New-EditorFile {
     end {
         # If editorContext is null, then we're in a Temp session and
         # this cmdlet won't work so return early.
-        $editorContext = $psEditor.GetEditorContext()
+        try {
+            $editorContext = $psEditor.GetEditorContext()
+        }
+        catch {
+            # If there's no editor, this throws an error. Create a new file, and grab the context here.
+            # This feels really hacky way to do it, but not sure if there's another way to detect editor context...
```

Comment:
This definitely appears like it would work, so I don't mind the approach. I would like to know if there's a way to test if an editor context exists (if so we can avoid exception handling), and if creating a new file on the workspace is the best way to generate a new editor context. If this is the only approach, then I feel like our interface is lacking because as someone new to the code, I would have expected something like:

* `GetEditorContext()` (which exists)
* `HasEditorContext()` (essentially says if the above would throw)
* `NewEditorContext()` (to create one, instead of relying what I presume is a side-effect of `Workspace.NewFile()`)